project/SparkBuild.scala

```diff
@@ -237,7 +237,14 @@ object SparkBuild extends PomBuild {
         //  fixed.
         "-Wconf:msg=^(?=.*?method|value|type|object|trait|inheritance)(?=.*?deprecated)(?=.*?since 2.13).+$:s",
         "-Wconf:msg=^(?=.*?Widening conversion from)(?=.*?is deprecated because it loses precision).+$:s",
-        "-Wconf:msg=Auto-application to \\`\\(\\)\\` is deprecated:s",
+        // SPARK-45610 Convert "Auto-application to `()` is deprecated" to compile error, as it will become a compile error in Scala 3.
+        "-Wconf:cat=deprecation&msg=Auto-application to \\`\\(\\)\\` is deprecated:e",
```

Comment:
As this compilation warning will become a compilation error in Scala 3, to ensure it does not occur again, this PR also converts it into a compilation error in Scala 2.13.

If we don't need to do this protection now, please let me know.

cc @srowen  FYI

Reply:
It seems OK, but I can't access the test result for some reason to see why this failed.

Reply:
Recently, the `Run / Build modules: streaming, sql-kafka-0-10, streaming-kafka-0-10, mllib-local, mllib, yarn, kubernetes, hadoop-cloud, spark-ganglia-lgpl, connect, protobuf` task is not easily successful, the compilation container is always getting killed, I have triggered it again.

Reply:
All test passed